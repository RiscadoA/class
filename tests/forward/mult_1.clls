proc main() {
    cut {
        println("recv x");
        recv a(x);
        par {
            println("close x");
            close x
            ||
            println("close a");
            close a
        }
        |a:send wait; wait|
        println("send y");
        send a(y. println("wait y"); wait y; println("end y"); ());
        println("forward");
        fwd a b
        |b:wait|
        println("wait b");
        wait b;
        println("end b");
        ()
    }
};;
