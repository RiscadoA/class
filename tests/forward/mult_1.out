send y
recv x
close x
wait y
end y
close a
forward
wait b
end b
